index.js

```diff
@@ -8,37 +8,40 @@ module.exports = function fetchCookieDecorator (fetch, jar) {
   var getCookieString = denodeify(jar.getCookieString.bind(jar))
   var setCookie = denodeify(jar.setCookie.bind(jar))
 
-  return function fetchCookie (url, opts) {
+  async function fetchCookie (url, opts) {
     opts = opts || {}
 
-    return getCookieString(url)
-      .then(function (cookie) {
-        return fetch(url, Object.assign(opts, {
-          headers: Object.assign(opts.headers || {}, (cookie ? { cookie: cookie } : {}))
-        }))
-      })
-      .then(function (res) {
-        var cookies
-
-        if (res.headers.getAll) {
-          // node-fetch v1
-          cookies = res.headers.getAll('set-cookie')
-        } else {
-          // node-fetch v2
-          var cookie = res.headers.get('set-cookie')
-          // FIXME: This is not working. See issue #22 (https://github.com/valeriangalliat/fetch-cookie/issues/22)
-          cookies = cookie.split(',') || []
-        }
-
-        if (!cookies.length) {
-          return res
-        }
-
-        return Promise.all(cookies.map(function (cookie) {
-          return setCookie(cookie, res.url)
-        })).then(function () {
-          return res
-        })
-      })
+    // Prepare request
+    const cookie = await getCookieString(url)
+    const headers = Object.assign(opts.headers || {}, (cookie ? { cookie: cookie } : {}))
+    opts = Object.assign(opts, {
+      headers: headers
+    })
+
+    // Actual request
+    const res = await fetch(url, opts)
+
+    // Get cookie header
+    var cookies = []
```

Comment:
Could be using `let` as we already use `const`

Reply:
True. Will change when refactoring or working on something else.